source/common/router/config_impl.h

```diff
@@ -170,27 +171,32 @@ class VirtualHostImpl : public VirtualHost {
   enum class SslRequirements { NONE, EXTERNAL_ONLY, ALL };
 
   struct VirtualClusterEntry : public VirtualCluster {
-    VirtualClusterEntry(const envoy::api::v2::route::VirtualCluster& virtual_cluster);
+    VirtualClusterEntry(const envoy::api::v2::route::VirtualCluster& virtual_cluster,
+                        Stats::StatNamePool& pool);
 
     // Router::VirtualCluster
-    const std::string& name() const override { return name_; }
+    Stats::StatName statName() const override { return stat_name_; }
 
     std::regex pattern_;
     absl::optional<std::string> method_;
-    std::string name_;
+    Stats::StatName stat_name_;
   };
 
-  struct CatchAllVirtualCluster : public VirtualCluster {
+  class CatchAllVirtualCluster : public VirtualCluster {
+  public:
+    explicit CatchAllVirtualCluster(Stats::StatNamePool& pool) : stat_name_(pool.add("other")) {}
+
     // Router::VirtualCluster
-    const std::string& name() const override { return name_; }
+    Stats::StatName statName() const override { return stat_name_; }
 
-    std::string name_{"other"};
+  private:
+    Stats::StatName stat_name_;
```

Comment:
nit: const here and elsewhere if possible.

Reply:
done